app/services/puppet_fact_parser.rb

```diff
@@ -5,9 +5,14 @@ def operatingsystem
     orel = os_release.dup
 
     if orel.present?
-      major, minor = orel.split('.', 2)
-      major = major.to_s.gsub(/\D/, '')
-      minor = minor.to_s.gsub(/[^\d\.]/, '')
+      if os_name =~ /ubuntu/i
+        major = os_major_version
+        minor = os_minor_version
```

Comment:
On a related note (but not necessarily in this PR): I'm still inclined to make this the default, not just for Ubuntu but for all. I'd be happy to accept a follow up PR that implements that.

Reply:
Yes, but this should definitely be handled by an extra PR as mentioned [here](https://github.com/theforeman/foreman/pull/9760#discussion_r1582992217)